shiny/experimental/ui/_sidebar.py

```diff
@@ -207,6 +208,9 @@ def sidebar(
     if id is None and open != "always":
         # but always provide id when collapsible for accessibility reasons
         id = f"bslib-sidebar-{random.randint(1000, 10000)}"
+    else:
+        if id:
+            id = resolve_id(id)
```

Comment:
The structure of this conditional feels weird to me. How about this? I think it should result int he same logic:

```py
if id is not None:
    ...
elif open != "always":
    ...
```

But also, now that I think about it, shouldn't `id` always go through `resolve_id()`, even if it started out as `None`?

Reply:
new 210 is needed and I like where it's at. But the else can be removed and we'll just use `resolve_id(id)`